src/run_program.rs

```diff
@@ -336,7 +336,7 @@ impl<'a, D: Dialect> RunProgramContext<'a, D> {
         let extension =
             self.dialect
                 .softfork_extension(uint_atom::<4>(self.allocator, extension, "softfork")? as u32);
-        if extension == OperatorSet::Default {
+        if extension == OperatorSet::Unknown {
```

Comment:
this failure must depend on whether we're in mempool or not, right? Do we ignore the "unknown softfork extension" error we issue below, in the caller, if we're in consensus mode?